singer-commons/src/main/thrift/loggingaudit_config.thrift

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2019 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace py schemas.loggingaudit
+namespace java com.pinterest.singer.loggingaudit.thrift.configuration
+
+include "loggingaudit.thrift"
+include "common.thrift"
+
+
+/**
+ *  AuditEvent sender type.
+ **/
+enum SenderType {
+   KAFKA = 0
+}
+
+/**
+ *  audit config for each topic/logstream.
+ **/
+struct AuditConfig{
+
+   /**
+    *  samplingRate specify the percent of log messages are going to be audited. This field needs to
+    *  be set in two cases:
+    *  (1) messages are logged by thrift logger and this field needs to be set only at ThriftLogger
+    *      stage, not the following stages such as Singer and Merced.
+    *  (2) Messages are not generated by thrift logger and Singer is the first stage of thet pipeline,
+    *      for example, Singer can be configured to upload server logs (in text format).
+    */
+    1: optional double samplingRate = 1.0;
+
+   /**
+    *  flag indicates whether current stage is the start of logging auditing. For ThriftLogger stage,
+    *  the value is always true. At Singer stage, this field can be true if Singer uploads log files
+    *  not generated by ThriftLogger, for example server logs in text format. For other
+    *  stages, this field is always false.
+    */
+    2: optional bool startAtCurrentStage = true;
+
+   /**
+    *  flag indicates whether current stage is the end of the logging auditing. This field can be
+    *  true at Merced / Merced_HR stage.
+    */
+    3: optional bool stopAtCurrentStage = false;
+}
+
+
+/**
+ *  KakfaSenderConfig consists of kafka topic name and kafka producer config.
+ **/
+struct KafkaSenderConfig {
+    1: optional string topic = "logging_audit";
+    2: optional common.KafkaProducerConfig kafkaProducerConfig;
+    3: optional i32 stopGracePeriodInSeconds = 300;
```

Comment:
change the default value to 30 seconds?  300 seconds can be too long  for java service restart.

Reply:
I made the change in the update of the PR